### ISSUE TITLE
dvc 1.10.2 does not provide the extra 'webdavs'

### ISSUE THREAD

**Issue by ANaumann85** [iterative/dvc#4987]
## Bug Report
I wanted to test the webdav protocol. The command `dvc push -r webdavremote` (where webdavremote is the name of my remote). That gave me the useful output:
```
ERROR: failed to push data to the cloud - URL '***snipped***' is supported but requires these missing dependencies: ['webdavclient3']. If you have installed dvc using pip, choose one of these options to proceed: 

    1) Install specific missing dependencies:
        pip install webdavclient3
    2) Install dvc package that includes those missing dependencies: 
        pip install 'dvc[webdavs]'
    3) Install dvc package with all possible dependencies included: 
        pip install 'dvc[all]'

If you have installed dvc from a binary package and you are still seeing this message, please report it to us using https://github.com/iterative/dvc/issues. Thank you!
```
But the installation of the webdav extension using pip3 leads to the following first lines:
```
pip3 install 'dvc[webdavs]'
Requirement already satisfied: dvc[webdavs] in ${HOME}/.local/lib/python3.7/site-packages (1.10.2)
  dvc 1.10.2 does not provide the extra 'webdavs'
Requirement already satisfied: dictdiffer>=0.8.1 in ${HOME}/.local/lib/python3.7/site-packages (from dvc[webdavs]) (0.8.1)
Requirement already satisfied: toml>=0.10.1 in ${HOME}/.local/lib/python3.7/site-packages (from dvc[webdavs]) (0.10.2)
```
Hence the installation did nothing. I am a bit puzzled, why there is the warning and at the same time the extra _dvc[webdavs]_ at the end.

### Please provide information about your setup

**Output of `dvc version`:**

```console
dvc version
DVC version: 1.10.2 (pip)
---------------------------------
Platform: Python 3.7.3 on Linux-4.19.0-12-amd64-x86_64-with-debian-10.6
Supports: All remotes
Cache types: hardlink, symlink
Caches: local
Remotes: local, webdavs
Repo: dvc, git
```
That line "Supports: All remotes" was printed, after I installed `dvc[all]`. 
**Additional Information (if any):**

Which additional information would be helpful?

**Comment by pmrowla** [comment]
The error message should say to install `dvc[webdav]` and not `[webdavs]`, but either way it should also be installed automatically with `[all]`.

Is this a DVC installation that you've upgraded from earlier versions via pip? webdav support was added (relatively) recently, so it's possible that if you upgraded via pip it was not installed.

Running
```
pip install --upgrade dvc
```
will only upgrade core DVC, and not any of the remote dependencies.

To upgrade DVC and its remote dependencies, you need to run
```
pip install --upgrade dvc[all]
```
(or substitute the specific remotes you wish to upgrade instead of `all`).

**Comment by ANaumann85** [comment]
Yes, I upgraded from 1.10.1 via pip. But I had not installed any remote bevore and run
```
pip3 install dvc[all]
```
which installed all remotes successfully. 

Hence, from your answer, I assume that the trailing s in _webdavs_ was the only problem?

**Comment by pmrowla** [comment]
Yes, the issue was just the trailing s. Internally, the scheme for webdav over SSL/TLS is `webdavs`, but the extra to be installed is `webdav` (which covers both `webdav` and `webdavs` remote URL schemes). However, when we generate the error message for a missing remote dependency we incorrectly substitute the URL scheme rather than the proper pip extras name.

This bug would also apply to `http` remotes, for HTTPS URL's the error message will suggest installing `dvc[https]` when it should be `dvc[http]`.

**Comment by efiop** [comment]
> This bug would also apply to http remotes, for HTTPS URL's the error message will suggest installing dvc[https] when it should be dvc[http].

Just for the record: http support is included into the default package and doesn't require any extras, so we don't have to fix it. Created #5005 to fix it for webdavs.
